mapie/quantile_regression.py

```diff
@@ -0,0 +1,425 @@
+
+        Raises
+        ------
+        ValueError
+            If the estimator fit or predict methods.
+        ValueError
+            We check if it's a known estimator that does quantile regression
+            according to the dictionnary set quantile_estimator_params.
+            This dictionnary will need to be updated with the latest new
+            available estimators.
+        ValueError
+            The estimator does not have the "loss_name" in its parameters and
+            therefore can not be used as an estimator.
+        ValueError
+            There is no quantile "loss_name" and therefore this estimator
+            can not be used as a ``MapieQuantileRegressor``.
+        ValueError
+            The parameter to set the alpha value does not exist in this
+            estimator and therefore we cannot use it.
+        """
+        if estimator is None:
+            return QuantileRegressor(
+                solver="highs-ds",
+                alpha=1e-9,
+            )
+        if not (hasattr(estimator, "fit") and hasattr(estimator, "predict")):
+            raise ValueError(
+                "Invalid estimator. "
+                "Please provide a regressor with fit and predict methods."
+            )
+        if isinstance(estimator, Pipeline):
+            self._check_estimator(estimator[-1])
+            return estimator
+        else:
+            name_estimator = estimator.__class__.__name__
+            if name_estimator == "QuantileRegressor":
+                return estimator
+            else:
+                if name_estimator in self.quantile_estimator_params:
+                    param_estimator = estimator.get_params()
+                    loss_name, alpha_name = self.quantile_estimator_params[
+                        name_estimator].values()
+                    if loss_name in param_estimator:
+                        if param_estimator[loss_name] != "quantile":
+                            raise ValueError(
+                                "You need to set the loss/objective argument"
+                                + " of your base model ``quantile``."
+                            )
+                        else:
+                            if alpha_name in param_estimator:
+                                return estimator
+                            else:
+                                raise ValueError(
+                                    "The matching parameter alpha_name for"
+                                    + " estimator does not exist."
+                                )
+                    else:
+                        raise ValueError(
+                            "The matching parameter loss_name for"
+                            + " estimator does not exist."
+                        )
+                else:
+                    raise ValueError(
+                        "The base model does not seem to be accepted"
+                        + " by MapieQuantileRegressor."
```

Comment:
Add a message saying :
- Give a base model among `quantile_estimator_params.keys()`
- Add your base model to `quantile_estimator_params`

Reply:
done